mods/uxtheme-hook.wh.cpp

```diff
@@ -0,0 +1,169 @@
+// @name            UXTheme hook
+// @description     Allows you to apply custom themes
+// @version         1.0
+// @author          rounk-ctrl
+// @github          https://github.com/rounk-ctrl
+// @include         winlogon.exe
+// @include         explorer.exe
+// @include         systemsettings.exe
+// @include         logonui.exe
+// @compilerOptions -luxtheme -lgdi32
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# UXTheme hook
+apply custom themes.
+
+### ⚠ Important ⚠
+in order for this mod to work properly, you must include `winlogon.exe` and `logonui.exe` in windhawk's process inclusion list, in its advanced settings.
+
+### bugs
+the colors in control panel might be weird sometimes. works fine if you leave control panel open, then switch themes.
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- cpanelFix: true
+  $name: Fix control panel white header/sidebar
+  $description: Fixes control panel hardcoding white image for header and sidebar
+*/
+// ==/WindhawkModSettings==
+
+#include <vssym32.h>
+#include <windhawk_utils.h>
+#include <vector>
+#include <uxtheme.h>
+
+typedef unsigned __int64 QWORD;
+
+struct {
+    bool cpanelFix;
+} settings;
+
+#pragma region common
+HRESULT (*CThemeSignature_Verify)(PVOID, PVOID);
+HRESULT CThemeSignature_VerifyHook(
+    PVOID pThis,
+    PVOID hFile
+)
+{
+    return 0;
+}
+#pragma endregion
+
+#pragma region DUI
+typedef VOID(* WINAPI Element_PaintBgT)(class Element*, HDC , class Value*, LPRECT, LPRECT, LPRECT, LPRECT);
+Element_PaintBgT Element_PaintBg;
+VOID WINAPI Element_PaintBgHook(class Element* This, HDC hdc, class Value* value, LPRECT pRect, LPRECT pClipRect, LPRECT pExcludeRect, LPRECT pTargetRect)
+{
+    //unsigned char byteValue = *(reinterpret_cast<unsigned char*>(value) + 8);
+    if ((int)(*(DWORD *)value << 26) >> 26 != 9 )
+    {
+        auto v44 = *((QWORD *)value + 1);
+        auto v45 = (v44+20)& 7;
+        // 6-> selection
+        // 3-> hovered stuff
+        // 4-> cpanel top bar and side bar (white image)
+        // 1-> some new cp page style (cp_hub_frame)
+        if (v45==4)
+        { 
+            HWND wnd = WindowFromDC(hdc);
+            HTHEME hTh = OpenThemeData(wnd, L"ControlPanel");
+            COLORREF clrBg;
+            GetThemeColor(hTh, 2, 0, TMT_FILLCOLOR, &clrBg);
+            HBRUSH SolidBrush = CreateSolidBrush(clrBg);
+            FillRect(hdc, pRect, SolidBrush);
+            DeleteObject(SolidBrush);
+            CloseThemeData(hTh);
+        }
+        else
+        {
+            Element_PaintBg(This, hdc, value, pRect, pClipRect, pExcludeRect, pTargetRect);
+        }
+    }
+    else
+    {
+        Element_PaintBg(This, hdc, value, pRect, pClipRect, pExcludeRect, pTargetRect);
+    }
+}
+#pragma endregion
+
+
+using SetSysColors_t = decltype(&SetSysColors);
+SetSysColors_t SetSysColors_orig;
+
+int SetSysColors_hook(int cElements, const INT *lpaElements, const COLORREF *lpaRgbValues)

```

Comment:
@rounk-ctrl missed this before merging:

Calling conventions are missing, which is likely to cause a crash for 32-bit targets. Please add calling conventions or, if the mod doesn't support 32-bit, limit it to 64-bit, which will also make compilation faster. See also:
https://github.com/ramensoftware/windhawk/wiki/Development-tips#hooks-and-calling-conventions